### DISCUSSION TITLE
Geters item from Collection to new Collection

### DISCUSSION THREAD

**Discussion by sbc640964** [laravel/framework#43250] [Ideas]
It happens to me a lot that I retrieve an organ from the collection and want to continue the collection methods on it.

I know I could have added the following methods with macro, but still I think it comes in handy with a lot of developers, so maybe it's worth getting into the core of the framework.

```php

$collection - collect([...])

$collection->getToCollect($key);
$collection->firstToCollect();
$collection->lastToCollect();
$collection->firstWhereToCollect($key, $operator, $value);
...
```

**Comment by henzeb** [comment]
As some of those are higher order messages, it would be nice to be able to do
```php
$collection->first->collect();
```
Likewise, it could be nice to chain higher order messages.

```php
$collection->sortByDesc->first;
```

**Comment by michael-rubel** [comment]
@sbc640964 

You can make a `recursive` collection of your content using [`laravel-collection-macros`](https://github.com/spatie/laravel-collection-macros#recursive) package or just [`pipe`](https://laravel.com/docs/9.x/collections#method-pipe) into the collection and collect the values you need precisely.
